### PR TITLE
Apply DelayScalar to get correct samples offset

### PR DESCRIPTION
**Pull request by da-wad**
SEG-Y rev.1 gives the following info on trace header bytes 215-216:

> Scalar to be applied to times specified in Trace Header bytes 95-114 to give the true time value in milliseconds. 
Scalar = 1, +10, +100, +1000, or +10,000. 
If positive, scalar is used as a multiplier; 
if negative, scalar is used as divisor. 
A value of zero is assumed to be a scalar value of 1.

This scalar appeared not to be applied by segyio, making PR which should fix that.

### CODE REVIEW

File: python/segyio/open.py [hunk 1]
```diff
@@ -172,8 +172,14 @@ def open(filename, mode="r", iline = 189,
     )
 
     try:
+        # Bytes 215-216
```
Comment (ErlendHaa) [inline python/segyio/open.py:175]: Noice? Maybe remove?

File: python/segyio/open.py [hunk 2]
```diff
@@ -172,8 +172,14 @@ def open(filename, mode="r", iline = 189,
...
+        delay_scalar = f.header[0][segyio.TraceField.ScalarTraceHeader]
+        if delay_scalar == 0:  # A value of zero is assumed to be a scalar value of 1.
+            delay_scalar = 1
+        elif delay_scalar < 0:  # if negative, scalar is used as divisor.
```
Comment (ErlendHaa) [inline python/segyio/open.py:179]: Maybe remove these comment too. When I first read it I thought you made this assumption, but while checking the spec I see this is the intended behaviour. And the second one just explaining what's going on in the next line of code anyways

**Review by ErlendHaa — COMMENTED** [review]
Nice catch! This is definitely what the spec says. Good contribution. Merge the 2 commits together remove some comments and we'll merge it

File: python/segyio/open.py (hunk 2 above)
Comment (da-wad) [reply python/segyio/open.py]: Cleaner now 👍

**Comment by dmlbarker** [comment]
I can confirm that this fixes segyio for a previously wrongly interpreted segy

**Review by ErlendHaa — APPROVED** [review]
LGTM

**Pull request merged**